src/writer/command_line.py

```diff
@@ -37,14 +41,32 @@ def main():
     absolute_app_path = _get_absolute_app_path(
         args.path) if args.path else None
     host = args.host if args.host else None
+    api_key = args.api_key if args.api_key else None
+
+    _perform_checks(command, absolute_app_path, host, enable_remote_edit, api_key)
+    api_key = _get_api_key(command, api_key)
+    _route(command, absolute_app_path, port, host, enable_remote_edit, enable_server_setup_hook, template_name, api_key)
+
+def _get_api_key(command, api_key: Optional[str]) -> Optional[str]:
+    if command in ("deploy") and api_key is None:
+        env_key = os.getenv("WRITER_API_KEY", None)
+        if env_key is not None and env_key != "": 
+            return env_key
+        else:
+            logging.info("An API key is required to deploy a Streamsync app.")
```

Comment:
Writer app, please can you do a search of `streamsync` to find any old references